admin/app/controllers/workarea/admin/content_blocks_controller.rb

```diff
@@ -52,6 +52,22 @@ def move
       head :ok
     end
 
+    def copy
+      new_block = @block.clone
+      new_block.content = @block.content
+
+      if new_block.save
+        flash[:success] = t('workarea.admin.content_blocks.flash_messages.copied')
+      else
+        flash[:error] = new_block.errors.to_a.to_sentence
+      end
+
+      redirect_to return_to || edit_content_path(
```

Comment:
we have `redirect_back_or` for handling this. 
https://github.com/workarea-commerce/workarea/blob/master/core/app/controllers/workarea/authentication.rb#L100